### PR TITLE
WIP: Move to jaeger.thrift

### PR DESCRIPTION
**Pull request by pavolloffay**
@vprithvi @yurishkuro could you please review?

### CODE REVIEW

**Comment by pavolloffay** [comment]
CLA signed

**Comment by yurishkuro** [comment]
Have a couple of issues with this change.
  1. Ideally, we want to switch the client to `jaeger.thrift` that supports kv logs in the data model natively. It's on our roadmap anyway (#140), as currently only the Node.js client uses `jaeger.thrift` data model.
  1. In lieu of the previous point, the next best thing we can do is to replicate the logic in the Go client  where log fields are encoded as a JSON string and stored as Annotation almost without data loss (only the field value types are lost). Then in the backend we already have logic to process such format correctly and reconstruct the log fields from JSON before storing in OT-compatible format.

File: jaeger-core/src/main/java/com/uber/jaeger/reporters/protocols/ThriftSpanConverter.java [hunk 1]
```diff
@@ -76,7 +77,32 @@
...
+        if (logData.getPayload() != null) {
+          value = logData.getPayload().toString();
+        } else if (logData.getFields() != null && !logData.getFields().isEmpty()) {
+          // special-case the "event" field which is similar to the semantics of a zipkin annotation
+          Object event = logData.getFields().get("event");
+          if (event != null && logData.getFields().size() == 1) {
+            value  = event.toString();
+          } else {
+            StringBuilder result = new StringBuilder();
+            for (Iterator<? extends Map.Entry<String, ?>> i = logData.getFields().entrySet().iterator(); i.hasNext(); ) {
```
Comment (adriancole) [inline jaeger-core/src/main/java/com/uber/jaeger/reporters/protocols/ThriftSpanConverter.java]: might want to comment that this is the existing practice in zipkin-go-opentracing.. we do need to put together an OT mapping page on the website.. something bas and I spoke of in the past.

Comment (pavolloffay) [inline jaeger-core/src/main/java/com/uber/jaeger/reporters/protocols/ThriftSpanConverter.java]: This is definitely a good idea :+1:

**Comment by pavolloffay** [comment]
This should be an intermediate fix. You persuade me to port this to jaeger.thrift.

**Comment by pavolloffay** [comment]
Closing in favour of #139

**Comment by pavolloffay** [comment]
https://github.com/uber/jaeger-client-java/pull/142 is moving zipkin's `SpanThriftConverter` to `jaeger-zipkin` it is probably going to be used only for sending data to zipkin.. We could apply this patch if there is an interest

**Pull request closed without merging**